### DISCUSSION TITLE
Setting a value to an MFnGenericAttribute plug

### DISCUSSION THREAD

**Discussion by MFong-sms** [Autodesk/maya-usd#3436] [Q&A]
I am writing some USD importing code and need to import a string value into an MFnGenericAttribute that is set to accept string data.  However, the normal MPlug::setValue( ) call results in a crash in Maya.  I also tried using the pxr::UsdMayaReadUtil::SetMayaAttr( ) but it logs an error that it cannot set the string value to the specified plug.  I googled for setting values to MFnGenericAttributes but there wasn't anything that showed how it is done.
Examples of how to do this would be greatly appreciated.

**Comment by hamedsabri** [comment]
Hi all, hope all is well.

What @MFong-sms brought up here relates to our current convertor code that doesn't handle kGenericAttribute types:

https://github.com/Autodesk/maya-usd/blob/dev/lib/mayaUsd/utils/converter.h

We are wondering how one could handle MFnData::kAny type?

Cheers
